release/rancher/rancher.go

```diff
@@ -71,3 +83,64 @@ func rancherImages(imagesURL string) (string, error) {
 	}
 	return string(images), nil
 }
+
+func CheckRancherDockerImage(tag string) error {
+	ctx := context.Background()
+	org := "rancher"
+	repo := "rancher"
+	archs := []string{"amd64", "arm64", "s390x"}
+	exists, err := docker.CheckImageArchs(ctx, org, repo, tag, archs)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return errors.New("image tag doesn't exist or doesn't have all architectures")
+	}
+	return nil
+}
+
+func CheckHelmChartVersion(tag string) error {
+	versions, err := rancherHelmChartVersions(rancherHelmRepositoryURL)
+	if err != nil {
+		return err
+	}
+	foundVersion := false
+	for _, version := range versions {
+		logrus.Debug("checking version " + version)
+		if tag == version {
+			logrus.Info("found chart for version " + version)
+			foundVersion = true
+			break
+		}
+	}
+	if !foundVersion {
+		return errors.New("failed to find chart for rancher app version " + tag)
+	}
+	return nil
+}
+
+func rancherHelmChartVersions(repoURL string) ([]string, error) {
+	httpClient := http.Client{Timeout: time.Second * 15}
```

Comment:
Let's create a function in this repo that creates a new HTTP Client that takes a configurable timeout. This same line is used in quite a few places now and having configurable timeouts is nice.

Reply:
Since we're just configuring the timeout, I don't think it's worth importing a package and calling a function to create a new client

Reply:
This line exists in a number of locations. Create the function in this package if it's only in this package. Otherwise a new package.